dto/src/main/java/ca/on/oicr/gsi/runscanner/dto/dragen/DragenWorkflowAnalysis.java

```diff
@@ -0,0 +1,76 @@
+package ca.on.oicr.gsi.runscanner.dto.dragen;
+
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.LinkedList;
+import java.util.List;
+
+public class DragenWorkflowAnalysis {
+  private List<DragenAnalysisUnit> analyses = new LinkedList<>();
+  private Instant completionTime;
+  private Instant startTime;
+
+  public DragenAnalysisUnit get(String sample, String lane, String index1, String index2) {
+    return get(sample, lane, new StringBuilder(index1).append("-").append(index2).toString());
+  }
+
+  public DragenAnalysisUnit get(String sample, int lane, String index1, String index2) {
+    return get(sample, lane, new StringBuilder(index1).append("-").append(index2).toString());
+  }
+
+  public DragenAnalysisUnit get(String sample, String lane, String index) {
+    return get(sample, Integer.parseInt(lane), index);
+  }
+
+  public DragenAnalysisUnit get(String sample, int lane, String index) {
+    return analyses
+        .stream()
+        .filter(a -> a.getSample().equals(sample) && a.getLane() == lane)
+        .findFirst()
+        .orElse(new DragenAnalysisUnit());
+  }
+
+  public DragenAnalysisUnit get(Path filePath) {
+    List<DragenAnalysisUnit> list =
+        analyses
+            .stream()
+            .filter(a -> a.getFiles().stream().anyMatch(f -> f.getPath().equals(filePath)))
+            .toList();
+    if (list.size() > 1) {
+      throw new IllegalStateException(
+          "Can't have more than one Analysis with same file " + filePath);
```

Comment:
This is allowing bad data to be added and then validating it later upon retrieval? Or not validating it at all if get isn't called after adding?

Reply:
done